FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Mutual_Auth_wolfSSL/DemoTasks/MutualAuthMQTTExample.c

```diff
@@ -434,19 +434,26 @@ static void prvMQTTDemoTask( void * pvParameters )
     }
 }
 /*-----------------------------------------------------------*/
-
 static void prvTLSConnect( NetworkCredentials_t * pxNetworkCredentials,
                            NetworkContext_t * pxNetworkContext )
 {
     BaseType_t xNetworkStatus;
 
     /* Set the credentials for establishing a TLS connection. */
-    pxNetworkCredentials->pRootCa = ( const unsigned char * ) democonfigROOT_CA_PEM;
-    pxNetworkCredentials->rootCaSize = sizeof( democonfigROOT_CA_PEM );
-    pxNetworkCredentials->pClientCert = ( const unsigned char * ) democonfigCLIENT_CERTIFICATE_PEM;
-    pxNetworkCredentials->clientCertSize = sizeof( democonfigCLIENT_CERTIFICATE_PEM );
-    pxNetworkCredentials->pPrivateKey = ( const unsigned char * ) democonfigCLIENT_PRIVATE_KEY_PEM;
-    pxNetworkCredentials->privateKeySize = sizeof( democonfigCLIENT_PRIVATE_KEY_PEM );
+
+    pxNetworkCredentials->pRootCa     = (const unsigned char*)democonfigROOT_CA_PEM;
+    pxNetworkCredentials->pClientCert = (const unsigned char*)democonfigCLIENT_CERTIFICATE_PEM;
+    pxNetworkCredentials->pPrivateKey = (const unsigned char*)democonfigCLIENT_PRIVATE_KEY_PEM;
+
+#if defined( democonfigCREDENTIALS_IN_BUFFER)
+    pxNetworkCredentials->rootCaSize     = XSTRLEN(democonfigROOT_CA_PEM);
```

Comment:
What is XTRLEN? Could we consider using strlen or sizeof?

Reply:
XSTRLEN is a macro used in wolfSSL and will be expanded to strlen.  Agree to replace it with strlen.